src/stable/build/index.ts

```diff
@@ -0,0 +1,230 @@
+#!/usr/bin/env -S npx tsx
+
+import { IOType } from 'child_process';
+import { join } from 'path';
+
+import {
+    experimentalMessageCli,
+    experimentalMessageDfxJson
+} from '#build/utils/experimental_message';
+import { runCommand as runBuildCommand } from '#commands/build/index';
+import { runCommand as runCleanCommand } from '#commands/clean';
+import { runCommand as runDevSetupCommand } from '#commands/dev/setup/index';
+import { runCommand as runDevTemplateCommand } from '#commands/dev/template';
+import { runCommand as runExtensionInstallCommand } from '#commands/extension/install';
+import { runCommand as runGenerateCommand } from '#commands/generate/index';
+import { runCommand as runNewCommand } from '#commands/new';
+import { runCommand as runVersionCommand } from '#commands/version';
+import { getCanisterConfig } from '#utils/get_canister_config';
+import { AZLE_ROOT } from '#utils/global_paths';
+import { CanisterConfig, Command, SubCommand } from '#utils/types';
+
+import { version as azleVersion } from '../../../package.json';
+
+export async function build(): Promise<void> {
+    const command = process.argv[2] as Command | undefined;
+
+    if (command === undefined) {
+        throw new Error(
+            `No command found when running azle. Running azle should start like this: azle [commandName]`
+        );
+    }
+
+    const ioType = process.env.AZLE_VERBOSE === 'true' ? 'inherit' : 'pipe';
+
+    if (command === 'extension') {
+        await handleExtensionCommand(ioType);
+
+        return;
+    }
+
+    if (command === 'dev') {
+        await handleDevCommand();
+
+        return;
+    }
+
+    if (command === 'build') {
+        await handleBuildCommand(ioType);
+
+        return;
+    }
+
+    if (command === '--version') {
+        runVersionCommand();
+
+        return;
+    }
+
+    if (command === 'clean') {
+        await runCleanCommand();
+
+        return;
+    }
+
+    if (command === 'new') {
+        await handleNewCommand();
+
+        return;
+    }
+
+    if (command === 'generate') {
+        await handleGenerateCommand();
+
+        return;
+    }
+
+    // @ts-ignore stable extension still runs this command TODO prevent that
```

Comment:
Let's have a design discussion about this

Reply:
Do we want to at least make sure that we check that they aren't expecting files to be uploaded? Much in the same way as the original

Reply:
Let's make a post install command instead